tests/metadata.rs

```diff
@@ -154,3 +156,84 @@ fn metadata() {
     let digest = receipts[4].digest().expect("GetCaller should return contract Id");
     assert_eq!(&contract_call, digest);
 }
+
+#[test]
+fn get_transaction_fields() {
+    let rng = &mut StdRng::seed_from_u64(2322u64);
+
+    let mut client = MemoryClient::default();
+
+    let gas_price = 1;
+    let gas_limit = 1_000_000;
+    let maturity = 50;
+    let height = 122;
+    let input = 10_000_000;
+
+    let params = ConsensusParameters::default();
+
+    let tx = TransactionBuilder::script(vec![], vec![])
+        .maturity(maturity)
+        .gas_price(gas_price)
+        .gas_limit(gas_limit)
+        .add_unsigned_coin_input(rng.gen(), rng.gen(), input, AssetId::zeroed(), maturity)
+        .finalize_checked(height, &params);
+
+    let inputs = tx.as_ref().inputs();
+    let outputs = tx.as_ref().outputs();
+    let _witnesses = tx.as_ref().witnesses();
+
+    let inputs_bytes: Vec<Vec<u8>> = inputs.iter().map(|i| i.clone().to_bytes()).collect();
+    let _outputs_bytes: Vec<Vec<u8>> = outputs.iter().map(|o| o.clone().to_bytes()).collect();
+
+    #[rustfmt::skip]
+    let cases = vec![
+        inputs_bytes[0].clone(),
+    ];
+
+    #[rustfmt::skip]
+    let script = vec![
+        Opcode::MOVI(0x20, 0x01),
+        Opcode::gtf(0x30, 0x00, GTFArgs::ScriptData),
```

Comment:
Why is gtf here lowercase?

Reply:
We should probably split and build a dedicated dsl for the tx scripts, but its currently centralized in `Opcode` that that one is a function: [Opcode::gtf](https://docs.rs/fuel-asm/latest/fuel_asm/enum.Opcode.html#method.gtf)

Reply:
Oh so that's just for converting from GTFArgs to Imm12